cmd/snap-update-ns/change_test.go

```diff
@@ -2441,37 +2441,31 @@ func (s *changeSuite) TestComplexPropagatingChanges(c *C) {
 	// - $SNAP/b/c
 	// - $SNAP/d
 	//
-	// The mount profile contains but two entries:
+	// but snapd's mount profile contains only two entries:
 	//
-	// 1) recursive-bind $SNAP/b/c -> $SNAP/a
-	// 2) recursive-bind $SNAP/d -> $SNAP/b
-	//
-	// NB: The notation "foo -> bar" can be read as "foo is a view of bar",
-	// and can be implemented by mount "--rbind bar foo".
+	// 1) recursive-bind $SNAP/a -> $SNAP/b/c  (ie, mount --rbind $SNAP/a $SNAP/b/c)
+	// 2) recursive-bind $SNAP/b -> $SNAP/d    (ie, mount --rbind $SNAP/b $SNAP/d)
 	//
 	// Both mount operations are performed under a substrate that is MS_SHARED.
 	// Therefore, due to the rules that decide upon propagation of bind mounts
 	// the propagation of the new mount entries is also shared. This is
-	// documented in https://lwn.net/Articles/159092/ (section 5b).
+	// documented in section 5b of
+	// https://www.kernel.org/doc/Documentation/filesystems/sharedsubtree.txt.
 	//
 	// Interactive experimentation shows that the following three mount points exist
 	// after this operation, as illustrated by findmnt:
 	//
+	// TARGET                                SOURCE         FSTYPE      OPTIONS
+	// ...
 	// └─/snap/test-snapd-layout/x1          /dev/loop1     squashfs    ro,nodev,relatime
 	//   ├─/snap/test-snapd-layout/x1/b/c    /dev/loop1[/a] squashfs    ro,nodev,relatime
 	//   └─/snap/test-snapd-layout/x1/d      /dev/loop1[/b] squashfs    ro,nodev,relatime
 	//     └─/snap/test-snapd-layout/x1/d/c  /dev/loop1[/a] squashfs    ro,nodev,relatime
 	//
 	// Note that after the first mount operation only one mount point is created, namely
-	// $SNAP/b/c -> $SNAP/a. The second recursive bind mount not only creates
-	// $SNAP/d -> $SNAP/b, but also replicates $SNAP/b/c -> $SNAP/a as
-	// $SNAP/d/c -> $SNAP/a.
-	//
-	// The test will simulate a refresh of the snap from revision x1 to revision
-	// x2. When this happens the mount profile associated with x1 must be undone
-	// and the mount profile associated with x2 must be constructed. Because
-	// ordering matters, let's first consider the order of construction of x1
-	// itself. Starting from nothing, apply x1 as follows:
```

Comment:
FYI, I didn't mean for you to remove the 'The test will simulate...' paragraph. Feel free to add it back if desired.